spec/services/preserved_object_handler_update_version_spec.rb

```diff
@@ -340,4 +334,370 @@
+            version: po.current_version,
+            size: 1,
+            endpoint: ep,
+            status: Status.ok,
+            last_audited: Time.current.to_i,
+            last_checked_on_storage: Time.current
+          )
+        end
+
+        it 'updates PreservedCopy last_audited' do
+          orig_timestamp = pc.last_audited
+          sleep 1
+          po_handler.update_version_after_validation
+          expect(pc.reload.last_audited).to be > orig_timestamp
+        end
+        it 'updates PreservedCopy last_checked_on_storage' do
+          orig_timestamp = pc.last_checked_on_storage
+          po_handler.update_version_after_validation
+          expect(pc.reload.last_checked_on_storage).to be > orig_timestamp
+        end
+
+        it 'calls #update_online_version with validated = true and Status.ok' do
+          expect(po_handler).to receive(:update_online_version).with(true, Status.ok).and_call_original
+          po_handler.update_version_after_validation
+          skip 'test is lame b/c we only indirectly show the effects of #update_online_version in #update_version specs'
+        end
+
+        it 'updates PreservedCopy status to "ok" if it was "moab_invalid"' do
+          pc.status = Status.invalid_moab
+          pc.save!
+          po_handler.update_version_after_validation
+          expect(pc.reload.status).to eq Status.ok
+        end
+      end
+
+      context 'when moab is invalid' do
+        let(:druid) { 'xx000xx0000' }
+        let(:storage_dir) { 'spec/fixtures/bad_root01/bad_moab_storage_trunk' }
+        let(:ep) { Endpoint.find_by(storage_location: storage_dir) }
+
+        before do
+          # add storage root with the invalid moab to the Endpoints table
+          Endpoint.find_or_create_by!(endpoint_name: 'bad_fixture_dir') do |endpoint|
+            endpoint.endpoint_type = Endpoint.default_storage_root_endpoint_type
+            endpoint.endpoint_node = Settings.endpoints.storage_root_defaults.endpoint_node
+            endpoint.storage_location = storage_dir
+            endpoint.recovery_cost = Settings.endpoints.storage_root_defaults.recovery_cost
+            # endpoint.preservation_policies = PreservationPolicy.default_preservation_policy
+          end
+          # these need to be in before loop so it happens before each context below
+          PreservedObject.create!(druid: druid, current_version: 2, preservation_policy: default_prez_policy)
+          PreservedCopy.create!(
+            preserved_object: po, # TODO: see if we got the preserved object that we expected
+            version: po.current_version,
+            size: 1,
+            endpoint: ep,
+            status: Status.ok,
+            last_audited: Time.current.to_i,
+            last_checked_on_storage: Time.current
+          )
+        end
+
+        it 'updates PreservedCopy last_audited' do
+          orig_timestamp = pc.last_audited
+          sleep 1
```

Comment:
same answer about sleeping in tests.  :-D  (I think I deleted @jmartin-sul 's comment, by accident.)